src/python-common/ceph/deployment/service_spec.py

```diff
@@ -273,10 +273,25 @@ def validate(self) -> None:
         if self.hosts and self.label:
             # TODO: a less generic Exception
             raise SpecValidationError('Host and label are mutually exclusive')
-        if self.count is not None and self.count <= 0:
-            raise SpecValidationError("num/count must be > 1")
-        if self.count_per_host is not None and self.count_per_host < 1:
-            raise SpecValidationError("count-per-host must be >= 1")
+        if self.count is not None:
+            try:
+                intval = int(self.count)
+            except (ValueError, TypeError):
+                raise SpecValidationError("num/count must be a numeric value")
+            if self.count != intval:
+                raise SpecValidationError("num/count must be an integer value")
+            if self.count <= 0:
+                raise SpecValidationError("num/count must be > 1")
```

Comment:
```suggestion
                raise SpecValidationError("num/count must be >= 1")
```
Since that's what the check is actually verifying

Reply:
Sure, but FWIW that message existed in that form prior to my change. :-)
While we are discussing this - do you know the reason for "num/count" rather than just "count"? Is "num" supported somewhere as an alias for count? I couldn't find it but I can't say I looked very hard.

Reply:
More on this: 
```
            if self.count_per_host < 1:
                raise SpecValidationError("count-per-host must be >= 1")
```
While I'm here should we make the checks consistent too? In one case we have `x<= 0` and in the other `x < 1` which may be why the error text was written inconsistently. This would be a new patch.

Reply:
done in follow up patches